test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/DeepInsertSaveResultTests.cs

```diff
@@ -0,0 +1,316 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="BulkUpdateSaveResultTests.cs" company="Microsoft">

```

Comment:
Could we have tests that call `DeepInsert` directly, demonstrating how the user would interact with the feature?

Reply:
I will add them in the `response` PR to test end to end.